yt/frontends/athena_pp/data_structures.py

```diff
@@ -278,16 +278,9 @@ def _set_code_unit_attributes(self):
             mylog.warning("Assuming 1.0 = 1.0 %s", cgs)
             setattr(self, "%s_unit" % unit, self.quan(1.0, cgs))
 
-    def set_code_units(self):
-        super(AthenaPPDataset, self).set_code_units()
```

Comment:
I'm copying here the logic that you implemented for the Athena frontend, [see here](https://bitbucket.org/yt_analysis/yt/pull-requests/2213/mks-code-unit-system-fixes-closes-1229/diff). This is the real fix for the Athena++ bug. `_set_code_unit_attributes` is called. It's unnecessary to subclass `set_code_units` here.